### PR TITLE
Legacy index version to target indices created before N-2

### PR DESCRIPTION
**Pull request by javanna**
Legacy index versions can only be read via the archive indices functionality. The supported versions are currently 5.x and 6.x. For 9.0, we are not going to add support for indices created in 7.x as archive indices. Rather we will allow reading from N - 2 directly from Lucene, without relying on archive indices.

IndexVersion#isLegacyIndexVersion is tied to the archive indices functionality and identifies index versions that can only be present in the cluster as archive.

This commit aligns isLegacyIndexVersion to only match versions before N - 2. It also replaces usages of the method that need to identify indices created before N - 1 and are not specific to archive indices.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/repositories/RepositoriesModule.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public RepositoriesModule(
         }
         if (preRestoreChecks.isEmpty()) {
             preRestoreChecks.add((snapshot, version) -> {
-                if (version.isLegacyIndexVersion()) {
+                if (version.before(IndexVersions.MINIMUM_COMPATIBLE)) {
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/repositories/RepositoriesModule.java:105]: Similar to the comment above, we will likely change this too to N - 2 later, but for now we should not allow indices created in N-2 to be restored.

File: x-pack/plugin/old-lucene-versions/src/internalClusterTest/java/org/elasticsearch/xpack/lucene/bwc/AbstractArchiveTestCase.java [hunk 2]
```diff
@@ -94,11 +94,7 @@ public IndexMetadata getSnapshotIndexMetaData(RepositoryData repositoryData, Sna
                             .put(
                                 IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(),
                                 metadata.settings()
-                                    .getAsVersionId(
-                                        "version",
-                                        IndexVersion::fromId,
-                                        IndexVersion.fromId(randomFrom(5000099, 6000099, 7000099))
-                                    )
+                                    .getAsVersionId("version", IndexVersion::fromId, IndexVersion.fromId(randomFrom(5000099, 6000099)))
```
Comment (javanna) [inline x-pack/plugin/old-lucene-versions/src/internalClusterTest/java/org/elasticsearch/xpack/lucene/bwc/AbstractArchiveTestCase.java:97]: version 7 here was added as part of the lucene 10 ugprade but it was premature, as archive indices did not support yet lucene 8 indices.

**Comment by drempapis** [comment]
LGTM :)

**Review by drempapis — APPROVED** [review]

**Review by tlrx — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/repositories/RepositoriesModule.java (hunk 1 above)
Comment (javanna) [reply server/src/main/java/org/elasticsearch/repositories/RepositoriesModule.java:107]: Done.

**Pull request merged**